front/app/modules/commercial/customizable_homepage_banner/admin/CallToActionButtonSettings.tsx

```diff
@@ -14,39 +14,62 @@ import {
 } from 'services/appConfiguration';
 
 interface SigninStateProps {
-  options: string[];
+  options: (CTASignedOutButton | CTASignedInButton)[];
   currentValue: string;
   signInState: 'not_signed_in' | 'signed_in';
+  labelMessage: MessageDescriptor;
```

Comment:
But it will require using this pattern. And for me, it looks a bit ugly

```js
        {
          signed_in: messages.signed_in,
          not_signed_in: messages.not_signed_in,
        }[signInState]
```

Reply:
Haha, how does this beauty look ugly? I think it's a super elegant way of using ifs. And it's used a lot in the front-end, so you better start liking it. 😁 

Anyway, since there are only two possible states (signed in and signed out), you could use a ternary operator as well: `signInState === 'signed_in' ? messages.signed_in : messages.signed_out`

Reply:
> so you better start liking it

:smile: 

Ok, will change it.